### PR TITLE
[dev-v2.6] Bump harvester-csi-driver to v0.1.16 and harvester-cloud-provider to v0.1.14

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
Issue: https://github.com/harvester/harvester/issues/3571

This PR includes the two version Bump
1. Bump harvester-csi-driver to v0.1.16
2. Bump harvester-cloud-provider to v0.1.14

### CODE REVIEW

**Review by guangbochen — DISMISSED** [review]
LGTM, thanks.

**Review by jiaqiluo — CHANGES_REQUESTED** [review]
hi @Vicente-Cheng ,   We are currently in the process of releasing rancher v2.6.11,  do we need the new charts to be in rancher v2.6.11?  If not, we will not merge this PR until Rancher v2.6.11 is released.

**Comment by guangbochen** [comment]
@jiaqiluo We plan to release the Harvester v1.1.2 this week(support both Rancher v2.6 and v2.7); wondering whether the chart releases need to be bonded to the Rancher releases? or can it be fetched and updated in sync as long as user get access to the chart URL.

**Comment by jiaqiluo** [comment]
hi @guangbochen, it can be fetched and updated in sync as long as user gets access to the chart URL. 
To make it work, we need to add those two new charts to the release/v2.6 and release/v2.7 branches.  
It looks right for this PR to go into the dev-v2.6 branch. After the chart is validated by QA( I am not sure if this is done by Rancher QA), we need another PR for bringing them to the release-v2.6 branch. 

ping @sowmyav27 @snasovich for attention.

File: packages/harvester/harvester-cloud-provider/package.yaml [hunk 1]
```diff
@@ -1,2 +1,2 @@
-url: https://github.com/harvester/charts/releases/download/harvester-cloud-provider-0.1.13/harvester-cloud-provider-0.1.13.tgz
-version: 100.0.3
+url: https://github.com/harvester/charts/releases/download/harvester-cloud-provider-0.1.14/harvester-cloud-provider-0.1.14.tgz
+version: 101.0.3
```
Comment (snasovich) [inline packages/harvester/harvester-cloud-provider/package.yaml:2]: Charts for Rancher v2.6 should be in 100.x.y range, so we should probably bump it to `100.0.4` instead.

Comment (snasovich) [inline packages/harvester/harvester-cloud-provider/package.yaml]: Charts for Rancher v2.6 should be in 100.x.y range, so we should probably bump it to `100.0.4` instead.

File: packages/harvester/harvester-cloud-provider/generated-changes/patch/Chart.yaml.patch
Comment (snasovich) [inline packages/harvester/harvester-cloud-provider/generated-changes/patch/Chart.yaml.patch:1]: The same concerns with removing patch files as in https://github.com/rancher/charts/pull/2450/files#r1129899162

**Review by snasovich — CHANGES_REQUESTED** [review]
Couple comments to address + need to rebase since charts got released last week and `release.yaml` is now new.

File: packages/harvester/harvester-cloud-provider/generated-changes/patch/Chart.yaml.patch
Comment (Vicente-Cheng) [reply packages/harvester/harvester-cloud-provider/generated-changes/patch/Chart.yaml.patch:1]: Thanks, @snasovich. 
So you mean for the dev-v2.6. we need to set as following?
` catalog.cattle.io/rancher-version: '>= 2.6.0-0 < 2.7.0-0' `

Comment (snasovich) [reply packages/harvester/harvester-cloud-provider/generated-changes/patch/Chart.yaml.patch:1]: @Vicente-Cheng , for `dev-v2.6` it should be as per already released v2.6 chart - https://github.com/rancher/charts/blob/release-v2.6/charts/harvester-csi-driver/100.0.3%2Bup0.1.14/Chart.yaml#L8
So `'>= 2.6.1-0 < 2.7.0-0'` as I guess Harvester is not available in 2.6.0.

File: packages/harvester/harvester-cloud-provider/package.yaml (hunk 1 above)
Comment (Vicente-Cheng) [reply packages/harvester/harvester-cloud-provider/package.yaml]: Updated. Thanks for the reminder!

File: packages/harvester/harvester-cloud-provider/generated-changes/patch/Chart.yaml.patch
Comment (Vicente-Cheng) [reply packages/harvester/harvester-cloud-provider/generated-changes/patch/Chart.yaml.patch:1]: Updated!
@snasovich could you help to review this PR again?
Thanks!

**Review by snasovich — APPROVED** [review]
LGTM

**Review by jiaqiluo — APPROVED** [review]

**Review by guangbochen — APPROVED** [review]
LGTM, @jiaqiluo @snasovich please help to merge the PR, thanks.

**Pull request merged**